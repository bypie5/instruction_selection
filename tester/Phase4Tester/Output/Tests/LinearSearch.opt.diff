--- SelfTestCases/LinearSearch.opt.out	2011-04-28 14:55:21.000000000 -0700
+++ ./Output/Tests/LinearSearch.opt.out	2020-03-13 21:48:00.000000000 -0700
@@ -1,16 +1,20 @@
-10
-11
-12
-13
-14
-15
-16
-17
-18
-9999
-0
-1
-1
-0
-55
+Error in /Users/brandonyi/Documents/School/CS179E/instruction_selection/tester/Phase4Tester/./Output/Tests/LinearSearch.opt.compiler.out line 27 column 1: "la": Too few or incorrectly formatted operands. Expected: la $t1,($t2)
+Error in /Users/brandonyi/Documents/School/CS179E/instruction_selection/tester/Phase4Tester/./Output/Tests/LinearSearch.opt.compiler.out line 36 column 10: "9999": operand is of incorrect type
+Error in /Users/brandonyi/Documents/School/CS179E/instruction_selection/tester/Phase4Tester/./Output/Tests/LinearSearch.opt.compiler.out line 72 column 14: "$t2": operand is of incorrect type
+Error in /Users/brandonyi/Documents/School/CS179E/instruction_selection/tester/Phase4Tester/./Output/Tests/LinearSearch.opt.compiler.out line 80 column 1: "(" is not a recognized operator
+Error in /Users/brandonyi/Documents/School/CS179E/instruction_selection/tester/Phase4Tester/./Output/Tests/LinearSearch.opt.compiler.out line 104 column 14: "$t4": operand is of incorrect type
+Error in /Users/brandonyi/Documents/School/CS179E/instruction_selection/tester/Phase4Tester/./Output/Tests/LinearSearch.opt.compiler.out line 112 column 1: "(" is not a recognized operator
+Error in /Users/brandonyi/Documents/School/CS179E/instruction_selection/tester/Phase4Tester/./Output/Tests/LinearSearch.opt.compiler.out line 121 column 14: "$t1": operand is of incorrect type
+Error in /Users/brandonyi/Documents/School/CS179E/instruction_selection/tester/Phase4Tester/./Output/Tests/LinearSearch.opt.compiler.out line 125 column 14: "$t4": operand is of incorrect type
+Error in /Users/brandonyi/Documents/School/CS179E/instruction_selection/tester/Phase4Tester/./Output/Tests/LinearSearch.opt.compiler.out line 141 column 1: "la": Too few or incorrectly formatted operands. Expected: la $t1,($t2)
+Error in /Users/brandonyi/Documents/School/CS179E/instruction_selection/tester/Phase4Tester/./Output/Tests/LinearSearch.opt.compiler.out line 145 column 1: "la": Too few or incorrectly formatted operands. Expected: la $t1,($t2)
+Error in /Users/brandonyi/Documents/School/CS179E/instruction_selection/tester/Phase4Tester/./Output/Tests/LinearSearch.opt.compiler.out line 150 column 1: "la": Too few or incorrectly formatted operands. Expected: la $t1,($t2)
+Error in /Users/brandonyi/Documents/School/CS179E/instruction_selection/tester/Phase4Tester/./Output/Tests/LinearSearch.opt.compiler.out line 157 column 14: "$t2": operand is of incorrect type
+Error in /Users/brandonyi/Documents/School/CS179E/instruction_selection/tester/Phase4Tester/./Output/Tests/LinearSearch.opt.compiler.out line 159 column 9: "2": operand is of incorrect type
+Error in /Users/brandonyi/Documents/School/CS179E/instruction_selection/tester/Phase4Tester/./Output/Tests/LinearSearch.opt.compiler.out line 167 column 1: "(" is not a recognized operator
+Error in /Users/brandonyi/Documents/School/CS179E/instruction_selection/tester/Phase4Tester/./Output/Tests/LinearSearch.opt.compiler.out line 175 column 1: "la": Too few or incorrectly formatted operands. Expected: la $t1,($t2)
+Error in /Users/brandonyi/Documents/School/CS179E/instruction_selection/tester/Phase4Tester/./Output/Tests/LinearSearch.opt.compiler.out line 190 column 8: "$t1": operand is of incorrect type
+Error in /Users/brandonyi/Documents/School/CS179E/instruction_selection/tester/Phase4Tester/./Output/Tests/LinearSearch.opt.compiler.out line 193 column 1: "la": Too few or incorrectly formatted operands. Expected: la $t1,($t2)
+Error in /Users/brandonyi/Documents/School/CS179E/instruction_selection/tester/Phase4Tester/./Output/Tests/LinearSearch.opt.compiler.out line 2 column 1: Symbol "empty_LS" not found in symbol table.
 
+Processing terminated due to errors.
