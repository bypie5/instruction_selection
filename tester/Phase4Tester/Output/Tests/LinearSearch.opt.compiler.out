.data
empty_LS

.text
jal Main
li $v0 10
syscall

Main:
li $a0 8
jal _heapAlloc
move $t0 $v0
bnez $t0 null1
la $a0 _str0
j _error
null1:
move $a0 $t0
li $a1 10

move $t0 $v0
move $a0 $t0
jal _print



LS.Start:
la $t9 $s0
sw $t9 0($sp)
move $s0 $a0
move $t0 $a1
move $a0 $s0
move $a1 $t0

move $a0 $s0

move $a0 9999
jal _print
move $a0 $s0
li $a1 8

move $t0 $v0
move $a0 $t0
jal _print
move $a0 $s0
li $a1 12

move $t0 $v0
move $a0 $t0
jal _print
move $a0 $s0
li $a1 17

move $t0 $v0
move $a0 $t0
jal _print
move $a0 $s0
li $a1 50

move $t0 $v0
move $a0 $t0
jal _print
li $v0 55
lw $s0 0($sp)



LS.Print:
move $t0 $a0
li $t1 1
while1_top:
lw $t2 4($t0)
slti $t2 $t1 $t2
beqz $t2 while1_end
lw $t2 0($t0)
bnez $t2 null2
la $a0 _str0
j _error
null2:
lw $t3 0($t2)
(Lt) UNDEFINED
bnez $t3 bounds1
la $a0 _str0
j _error
bounds1:
mul $t3 $t1 4
addu $t3 $t3 $t2
lw $t3 4($t3)
move $a0 $t3
jal _print
addu $t1 $t1 1
j while1_top
while1_end:
li $v0 0



LS.Search:
move $t0 $a0
move $t1 $a1
li $t2 1
li $t3 0
while2_top:
lw $t4 4($t0)
slti $t4 $t2 $t4
beqz $t4 while2_end
lw $t4 0($t0)
bnez $t4 null3
la $a0 _str0
j _error
null3:
lw $t5 0($t4)
(Lt) UNDEFINED
bnez $t5 bounds2
la $a0 _str0
j _error
bounds2:
mul $t5 $t2 4
addu $t5 $t5 $t4
lw $t5 4($t5)
addu $t4 $t1 1
slti $t6 $t5 $t1
beqz $t6 if1_else
j if1_end
if1_else:
slti $t4 $t5 $t4
bnez $t4 if2_else
j if2_end
if2_else:
li $t3 1
lw $t2 4($t0)
if2_end:
if1_end:
addu $t2 $t2 1
j while2_top
while2_end:
move $v0 $t3



LS.Init:
la $t9 $s0
sw $t9 0($sp)
move $s0 $a0
move $t0 $a1
la $t9 $t0
sw $t9 4($s0)
move $a0 $t0

move $t0 $v0
la $t9 $t0
sw $t9 0($s0)
li $t0 1
lw $t1 4($s0)
addu $t1 $t1 1
while3_top:
lw $t2 4($s0)
slti $t2 $t0 $t2
beqz $t2 while3_end
mul $t2 2 $t0
subu $t3 $t1 3
lw $t4 0($s0)
bnez $t4 null4
la $a0 _str0
j _error
null4:
lw $t5 0($t4)
(Lt) UNDEFINED
bnez $t5 bounds3
la $a0 _str0
j _error
bounds3:
mul $t5 $t0 4
addu $t5 $t5 $t4
addu $t3 $t2 $t3
la $t9 $t3
sw $t9 4($t5)
addu $t0 $t0 1
subu $t1 $t1 1
j while3_top
while3_end:
li $v0 0
lw $s0 0($sp)



AllocArray:
move $t0 $a0
mul $t1 $t0 4
addu $t1 $t1 4
li $a0 $t1
jal _heapAlloc
move $t1 $v0
la $t9 $t0
sw $t9 0($t1)
move $v0 $t1




_print:
li $v0 1
syscall
la $a0 _newline
li $v0 4
syscall
jr $ra

_error:
li $v0 4
syscall
li $v0 10
syscall

_heapAlloc:
li $v0 9
syscall
jr $ra

.data
.align 0
_newline: .asciiz "\n"
_str0: .asciiz "null pointer\n"
